components/FilterableNumbersList.js

```diff
@@ -1,18 +1,47 @@
 import * as React from 'react';
 import { NumbersList } from './NumbersList';
+import { withRouter } from 'next/router';
+import { DebounceInput } from 'react-debounce-input';
+
+class _FilterableNumbersList extends React.Component {
+  state = {
+    query: this.props.router.query.q
```

Comment:
By passing in the query as an initial state, we make sure that users can go directly to a search url like http://localhost:3000/?q=888 and that the search box will reflect that state